packages/dashboard/src/components/queryEditor/advancedSearch/advancedSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Button from '@cloudscape-design/components/button';
+import FormField from '@cloudscape-design/components/form-field';
+import Input from '@cloudscape-design/components/input';
+import { useState, useMemo, useCallback } from 'react';
+import { ExecuteQueryCommandOutput } from '@aws-sdk/client-iottwinmaker';
+import { createKnowledgeGraphQueryClient } from '@iot-app-kit/react-components/src/components/knowledge-graph/KnowledgeGraphQueries';
+import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
+import { AdvancedResultsTable } from './advancedResultsTable';
+import { Property } from './model/property';
+
+export interface AdvancedSearchProps {
+  kGDatamodule: TwinMakerKGQueryDataModule;
+  onSelect: (properties: Property[]) => void;
+}
+
+/*
+    The return types are incorrect for Knowledge Graph
+    when selecting an edge instead of a node
+    so we can't properly type here.
+
+    When SiteWise sync is turned on with Knowledge Graph
+    it adds a definition field for the propertyId and displayName
+    and adds propertyName as a field.
+  */
+function executeQueryToProperty(res: any) {
+  const properties: Property[] = [];
+
+  if (res.rows) {
+    for (const row of res.rows) {
+      if (row.rowData) {
+        for (const data of row.rowData) {
+          if (data && data.definition) {
+            properties.push({
+              propertyId: data.definition.configuration.sitewisePropertyId,
+              displayName: data.definition.displayName ? data.definition.displayName : '-',
+              propertyName: data.propertyName,
+            });
+          }
+        }
+      }
+    }
+  }
+
+  return properties;
+}
+
+/** Advanced search for exploring your AWS IoT SiteWise properties. */
+export function AdvancedSearch({ kGDatamodule, onSelect }: AdvancedSearchProps) {
+  const [searchTermInput, setSearchTermInput] = useState('');
+  const [properties, setProperties] = useState<Property[]>([]);
+
+  const kGResponse = (res: ExecuteQueryCommandOutput) => {
+    setProperties(executeQueryToProperty(res));
+  };
+
+  const knowledgeGraphQueryClient = useMemo(() => {
+    return createKnowledgeGraphQueryClient(kGDatamodule, kGResponse);
+  }, [kGDatamodule, kGResponse]);
+
+  const onSearchClicked = useCallback(() => {
+    if (searchTermInput) {
+      knowledgeGraphQueryClient.findPropertiesByEntityOrPropertyNameOrDisplayName(searchTermInput);
+    }
+  }, [knowledgeGraphQueryClient, searchTermInput]);
```

Comment:
What is `useCallback` being utilized here for?

Reply:
I don't have a strong reason, in the `KnowledgeGraphPanel` the client was being used like this so I copied it. It stores the response so if the same request is made again it won't re-request the data. If there's any reason not to use it I can change it.